cmake/CeleritasAddTest.cmake

```diff
@@ -308,12 +308,49 @@ function(celeritas_add_test SOURCE_FILE)
 
     # Create an executable and link libraries against it
     add_executable(${_TARGET} "${SOURCE_FILE}" ${PARSE_SOURCES})
+    set_target_properties(${_TARGET} PROPERTIES
+      CUDA_SEPARABLE_COMPILATION ON
+      CUDA_RUNTIME_LIBRARY Shared
+    )
+
+    if (BUILD_SHARED_LIBS AND CELERITAS_USE_CUDA)
```

Comment:
Minor formatting consistency issue: `if` in CMake is a built-in function-like call, so CMake code shouldn't have a space after `if` or `endif` etc. (This takes some getting used to but ends up being more consistent internally and more consistent with the CMake code in official cmake modules.)